commerce_canadapost.module

```diff
@@ -22,26 +22,78 @@ const COMMERCE_CANADAPOST_LOGGER_CHANNEL = 'commerce_canadapost';
  */
 
 /**
- * Implements hook_entity_base_field_info().
- *
- * Adds a field to the shipment entity for holding the tracking PIN number.
+ * Implements hook_form_FORM_ID_alter().
  */
-function commerce_canadapost_entity_base_field_info(EntityTypeInterface $type) {
-  if ($type->id() !== 'commerce_shipment') {
+function commerce_canadapost_form_commerce_shipment_canadapost_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  array_unshift($form['actions']['submit']['#submit'],'commerce_canadapost_commerce_shipment_form_submit');
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function commerce_canadapost_form_commerce_checkout_flow_multistep_default_alter(&$form, FormStateInterface $form_state, $form_id) {
+  if(!isset($form['shipping_information']['shipments'])) {
+    return;
+  }
+
+  foreach ($form['shipping_information']['shipments'] as &$shipment) {
+    if (isset($shipment['field_actual_delivery'])) {
+      $shipment['field_actual_delivery']['#access'] = FALSE;
+    }
+    if (isset($shipment['field_attempted_delivery'])) {
+      $shipment['field_attempted_delivery']['#access'] = FALSE;
+    }
+    if (isset($shipment['field_expected_delivery'])) {
+      $shipment['field_expected_delivery']['#access'] = FALSE;
+    }
+    if (isset($shipment['field_mailed_on'])) {
+      $shipment['field_mailed_on']['#access'] = FALSE;
+    }
+  }
+}
```

Comment:
**NOT A CHANGE REQUEST**

*Readability*

No need to change this, but I would do the following to make this more readable:

```
foreach ($form['shipping_information']['shipments'] as &$shipment) {
  $fields = [
    'field_actual_delivery',
    'field_attempted_delivery',
    'field_expected_delivery',
    'field_mailed_on',
  ];
  foreach ($fields as $field) {
    if (isset($shipment[$field])) {
      $shipment[$field]['#access'] = FALSE;
    }
  }
}
```

It does add an extra level of indentation but overall it's clearer, especially since it's very clear at a glance that you're performing the same action on all those fields.